ethers-solc/src/resolver.rs

```diff
@@ -0,0 +1,518 @@
+                let component = component.unwrap();
+                if component == Component::CurDir || component == Component::ParentDir {
+                    // if the import is relative we assume it's already part of the processed input
+                    // file set
+                    match utils::canonicalize(node_dir.join(import)) {
+                        Ok(target) => {
+                            // the file at least exists,
+                            if let Some(idx) = index.get(&target).cloned() {
+                                resolved_imports.push(idx);
+                            } else {
+                                // imported file is not part of the input files
+                                let node = read_node(&target)?;
+                                unresolved.push_back((target.clone(), node));
+                                let idx = index.len();
+                                index.insert(target.clone(), idx);
+                                resolved_imports.push(idx);
+                            }
+                        }
+                        Err(err) => {
+                            tracing::trace!("failed to resolve relative import \"{:?}\"", err);
+                        }
+                    }
+                } else {
+                    // resolve library file
+                    if let Some(lib) = paths.resolve_library_import(import.as_ref()) {
+                        if let Some(idx) = index.get(&lib).cloned() {
+                            resolved_imports.push(idx);
+                        } else {
+                            // imported file is not part of the input files
+                            let node = read_node(&lib)?;
+                            unresolved.push_back((lib.clone(), node));
+                            let idx = index.len();
+                            index.insert(lib.clone(), idx);
+                            resolved_imports.push(idx);
+                        }
+                    } else {
+                        tracing::trace!(
+                            "failed to resolve library import \"{:?}\"",
+                            import.display()
+                        );
+                    }
+                }
+            }
+            nodes.push(node);
+            edges.push(resolved_imports);
+        }
+
+        Ok(Graph { nodes, edges, indices: index, num_input_files, root: paths.root.clone() })
+    }
+
+    /// Resolves the dependencies of a project's source contracts
+    pub fn resolve(paths: &ProjectPathsConfig) -> Result<Graph> {
+        Self::resolve_sources(paths, paths.read_input_files()?)
+    }
+}
+
+#[cfg(all(feature = "svm", feature = "async"))]
+impl Graph {
+    /// Returns all input files together with their appropriate version.
+    ///
+    /// First we determine the compatible version for each input file (sources and test, see
+    /// `Self::resolve`) Then we also add any other dependencies
```

Comment:
```suggestion
    /// `Self::resolve`) Then we also add any other dependencies.
```

(or there was some unfinished sentence?)